share/spack/setup-env.sh

```diff
@@ -200,26 +200,40 @@ _spack_shell_wrapper() {
 # or   pathadd OTHERPATH /path/to/dir  # add to OTHERPATH
 ########################################################################
 _spack_pathadd() {
-    # If no variable name is supplied, just append to PATH
-    # otherwise append to that variable.
+    # If no variable name is supplied, just prepend to PATH
+    # otherwise prepend to that variable.
+    # If the value is already in the variable, move it to the front
     _pa_varname=PATH
     _pa_new_path="$1"
     if [ -n "$2" ]; then
         _pa_varname="$1"
         _pa_new_path="$2"
     fi
 
-    # Do the actual prepending here.
+    # Get the current value of the variable
+    # This is equivalent to the bashism
+    #      _pa_oldvalue=${!_pa_varname}
+    # but supports all POSIX shells
     eval "_pa_oldvalue=\${${_pa_varname}:-}"
 
-    _pa_canonical=":$_pa_oldvalue:"
-    if [ -d "$_pa_new_path" ] && \
-       [ "${_pa_canonical#*:${_pa_new_path}:}" = "${_pa_canonical}" ];
+    # do nothing if the new path doesn't exist
+    if [ -d "$_pa_new_path" ];
```

Comment:
This check does not appear to be consistent with the comment at lie 219.

Reply:
I'm a little confused by this. The code is gated by a `-d` check to confirm that the new path exists as a directory.

Reply:
Ahhh!  My bad.  I read a negative comment -- `do nothing if does not exist` -- and let that influence my comment.

Can you change the comment to be more reflective of the check (i.e., in a positive way).

Reply:
I'll take the lack of a comment change as a "no".